samples/native_training/main.c

```diff
@@ -0,0 +1,221 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <stdio.h>
+
+#include "iree/runtime/api.h"
+
+float w[] = {4.0f, 4.0f, 5.0f};
```

Comment:
just as a matter of code hygiene, can you put these into a "host copy of state" struct and pass around a pointer to it? (even if you still have a global instance of it). Following code with global variables is hard.

Reply:
Done. I would actually prefer (and tried) just passing around non-global arrays, but it doesn't work with `IREE_ARRAYSIZE`.  But it looks like the struct approach works.